ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -798,9 +799,21 @@ impl<'a> TraceBuilder<'a> {
                 let bb = case_dests[cidx];
 
                 // Build the constant value to guard.
-                let jit_const = jit_int_type
-                    .to_owned()
-                    .make_constant(&mut self.jit_mod, val)?;
+                let jit_const = match jit_int_type.num_bits() {
+                    8 => jit_int_type
+                        .to_owned()
+                        .make_constant(&mut self.jit_mod, u8::try_from(val).unwrap())?,
+                    16 => jit_int_type
+                        .to_owned()
+                        .make_constant(&mut self.jit_mod, u16::try_from(val).unwrap())?,
+                    32 => jit_int_type
+                        .to_owned()
+                        .make_constant(&mut self.jit_mod, u32::try_from(val).unwrap())?,
+                    64 => jit_int_type
+                        .to_owned()
+                        .make_constant(&mut self.jit_mod, u64::try_from(val).unwrap())?,
+                    _ => unreachable!(),
```

Comment:
Is that unreachable or `todo!()`? I think you can switch on any arbitrary integer type.

Not that it matters much, they both stop the program with a crash.

Reply:
Yes, they're effectively equivalent as they both panic: at the moment this branch is literally unreachable, if I read ykllvm's serialiser correctly.

Reply:
I think it should be theoretically possible, but let's just merge it for now.